dispatcher/dispatcher.go

```diff
@@ -16,28 +17,39 @@ var dispatchers roster.Roster
 // DispatcherInitializationFunc is a function used to initialize an implementation of the `webhookd.WebhookDispatcher` interface.
 type DispatcherInitializationFunc func(ctx context.Context, uri string) (webhookd.WebhookDispatcher, error)
 
+// ctxUrl is a `context.Context` key used to store the URL that a `HTTPDispatcher` instance is configured to send messages to.
+type ctxUrl struct{}
+
+// ctxMethod is a `context.Context` key used to store the HTTP method that a `HTTPDispatcher` instance is configured to use.
+type ctxMethod struct{}
+
 // NewDispatcher() returns a new `webhookd.WebhookDispatcher` instance derived from 'uri'. The semantics of and requirements for
 // 'uri' as specific to the package implementing the interface.
 func NewDispatcher(ctx context.Context, uri string) (webhookd.WebhookDispatcher, error) {
 
 	err := ensureDispatcherRoster()
 
 	if err != nil {
-		return nil, fmt.Errorf("Failed to ensure dispatcher roster, %w", err)
+		return nil, fmt.Errorf("failed to ensure dispatcher roster, %w", err)
 	}
 
 	parsed, err := url.Parse(uri)
 
 	if err != nil {
-		return nil, fmt.Errorf("Failed to parse URI, %w", err)
+		return nil, fmt.Errorf("failed to parse URI, %w", err)
 	}
 
 	scheme := parsed.Scheme
 
 	i, err := dispatchers.Driver(ctx, scheme)
+	if len(parsed.Query()) > 0 {
```

Comment:
These details should be handled in specific implementations of the `Dispatcher` interface (like `HTTPDispatcher`) not in the code that resolves URIs to individual dispatchers.

Reply:
Moved this logic into HTTPDispatcher